src/useKeyDown/useKeyDown.ts

```diff
@@ -0,0 +1,52 @@
+import { ComputedRef, computed, onUnmounted, reactive, unref } from 'vue'
+import { MaybeArray, MaybeRef } from '@/types/maybe'
+import { asArray } from '@/utilities/arrays'
+
+export type UseKeyDown = {
+  down: ComputedRef<boolean>,
+}
+
+export type UseKeyDownCallback = (event: KeyboardEvent) => void
+
+export type UseKeyDownArgs = [key: MaybeRef<MaybeArray<string>>, callback?: UseKeyDownCallback]
+
+function useKeyDownFactory(): (...args: UseKeyDownArgs) => UseKeyDown {
+  const downKeys = reactive<Set<string>>(new Set())
+  const callbacks = new Set<UseKeyDownCallback>()
+
+  function keyDownCallback(event: KeyboardEvent): void {
+    downKeys.add(event.key)
+
+    callbacks.forEach(callback => callback(event))
+  }
+
+  function keyUpCallback(event: KeyboardEvent): void {
+    downKeys.delete(event.key)
+  }
+
+  document.addEventListener('keydown', keyDownCallback)
+  document.addEventListener('keyup', keyUpCallback)
+
+  return (...[key, callback]: UseKeyDownArgs): UseKeyDown => {
+    const keys = computed(() => asArray(unref(key)))
+    const down = computed(() => keys.value.some(key => downKeys.has(key)))
+
+    const filteredCallback: UseKeyDownCallback = (event) => {
+      if (keys.value.includes(event.key)) {
+        callback?.(event)
+      }
+    }
+
+    callbacks.add(filteredCallback)
+
+    onUnmounted(() => {
+      callbacks.delete(filteredCallback)
+    })
```

Comment:
You could probably wrap this in a current instance check and make this usable outside a component. 

I think the only caveat is you might want to expose a `disconnect` handler since this is missing teardown of the event listeners and this is the only way to remove the callback otherwise